agents/wiregrid_encoder/wiregrid_encoder.py

```diff
@@ -0,0 +1,356 @@
+                        or (len(pru_clock)
+                            and (current_time - time_encoder_published)
+                            > SEC_ENCODER_TO_PUBLISH):
+                        enc_rdata['timestamps'] = received_time_list
+                        enc_rdata['data']['quadrature'] =\
+                            quad_data[::COUNTER_INFO_LENGTH]
+                        enc_rdata['data']['pru_clock'] =\
+                            pru_clock[::COUNTER_INFO_LENGTH]
+                        enc_rdata['data']['reference_degree'] =\
+                            (np.array(ref_count)[::COUNTER_INFO_LENGTH]
+                             * 360 / COUNTS_ON_BELT).tolist()
+                        enc_rdata['data']['error'] =\
+                            error_flag[::COUNTER_INFO_LENGTH]
+
+                        enc_rdata['data']['rotation_speed'] = rot_speed  # Hz
+                        self.agent.publish_to_feed(
+                            'wgencoder_rough', enc_rdata)
+                        # store session.data
+                        field_dict = {
+                            'quadrature': enc_rdata['data']['quadrature'],
+                            'pru_clock': enc_rdata['data']['pru_clock'],
+                            'reference_degree':
+                                enc_rdata['data']['reference_degree'],
+                            'error': enc_rdata['data']['error'],
+                            }
+                        session.data['timestamps'] = received_time_list
+                        session.data['fields'] = field_dict
+
+                        enc_fdata['timestamps'] =\
+                            count2time(pru_clock, received_time_list[0])
+                        enc_fdata['data']['quadrature'] = quad_data
+                        enc_fdata['data']['pru_clock'] = pru_clock
+                        enc_fdata['data']['reference_count'] = ref_count
+                        enc_fdata['data']['error'] = error_flag
+                        self.agent.publish_to_feed('wgencoder_full', enc_fdata)
+
+                        quad_data = []
+                        pru_clock = []
+                        ref_count = []
+                        error_flag = []
+                        received_time_list = []
+
+                        dclock = []
+                        dcount = []
+                        rot_speed = []
+
+                        time_encoder_published = current_time
+
+                        time.sleep(SLEEP)
+                        pass
+                    pass
+
+                with open('/data/wg-data/position.log', 'w') as f:
+                    f.write(str(shared_time)+' '+str(shared_position)+'\n')
+                    f.flush()
+                    pass
+                pass
+
+        self.agent.feeds['wgencoder_rough'].flush_buffer()
+        # This buffer (full data) has huge data size.
+        # self.agent.feeds['wgencoder_full'].flush_buffer()
```

Comment:
Does running this flush_buffer cause issues?

Reply:
In the past, I and Hironobu faced the problem of increase of memory usage in this agent.
We removed it to avoid it just in case the flush_buffer caused some problem.
But, I don't know if it really causes it or not.

I think you know this problem a bit because Hironobu asked you about it.

Now, temporarily I added this flush_buffer() to check if it really makes any problem or not again.

Reply:
Gotcha, let me know how it goes, I won't let it hold up the PR, we can change things later if it proves to be an issue.